plugins/asus-hid/fu-asus-hid-device.c

```diff
@@ -85,35 +84,94 @@ fu_asus_hid_device_child_added_cb(FuDevice *device, FuDevice *child, gpointer us
 	fu_device_set_proxy(child, device);
 }
 
+static gboolean
+fu_asus_hid_device_validate_descriptor(FuDevice *device, GError **error)
+{
+#ifdef HAVE_HIDRAW_H
+	gint desc_size = 0;
+	struct hidraw_report_descriptor rpt_desc = {0x0};
+	g_autoptr(FuDevice) hid_device = NULL;
+	g_autoptr(FuFirmware) descriptor = fu_hid_descriptor_new();
+	g_autoptr(FuHidReport) report = NULL;
+	g_autoptr(FuIoctl) ioctl = fu_udev_device_ioctl_new(FU_UDEV_DEVICE(device));
+	g_autoptr(GBytes) fw = NULL;
+	g_autoptr(GPtrArray) imgs = NULL;
+
+	/* Get Report Descriptor Size */
+	if (!fu_ioctl_execute(ioctl,
+			      HIDIOCGRDESCSIZE,
+			      (guint8 *)&desc_size,
+			      sizeof(desc_size),
+			      NULL,
+			      5000,
+			      FU_IOCTL_FLAG_NONE,
+			      error))
+		return FALSE;
+
+	rpt_desc.size = desc_size;
+	if (!fu_ioctl_execute(ioctl,
+			      HIDIOCGRDESC,
+			      (guint8 *)&rpt_desc,
+			      sizeof(rpt_desc),
+			      NULL,
+			      5000,
+			      FU_IOCTL_FLAG_NONE,
+			      error))
+		return FALSE;
+	fu_dump_raw(G_LOG_DOMAIN, "HID descriptor", rpt_desc.value, rpt_desc.size);
+
+	fw = g_bytes_new(rpt_desc.value, rpt_desc.size);
+	if (!fu_firmware_parse_bytes(descriptor, fw, 0x0, FU_FIRMWARE_PARSE_FLAG_NONE, error)) {
+		return FALSE;
+	}
```

Comment:
Given this is exactly what `fu_pxi_ble_device_check_support_report_id()` is also doing, I think it makes sense to split this out in common code to use here. e.g. add something like `FuHidDescriptor *fu_hidraw_device_parse_descriptor(FuHidrawDevice *self, GError **error)` -- it also moves the HIDRAW define and header to where I think it belongs.